pkg/plugins/autodiscovery/flux/ocirepository.go

```diff
@@ -0,0 +1,48 @@
+package flux
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// https://fluxcd.io/flux/components/source/ocirepositories/#writing-an-ocirepository-spec
+
+type ociRepository struct {
+	ApiVersion string            `yaml:"apiVersion,omitempty"`
+	Kind       string            `yaml:"kind,omitempty"`
+	Metadata   map[string]string `yaml:"metadata,omitempty"`
+	Spec       ociRepositorySpec `yaml:"spec,omitempty"`
+}
+
+type ociRepositorySpec struct {
+	URL string               `yaml:"url,omitempty"`
+	Ref ociRepositorySpecRef `yaml:"ref,omitempty"`
+}
+
+type ociRepositorySpecRef struct {
+	Tag string `yaml:"tag,omitempty"`
+}
+
+func loadOCIRepository(filename string) (*ociRepository, error) {
+	var ociRepository ociRepository
+	var data []byte
+
+	data, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("opening file %s: %s", filename, err.Error())
+	}
+
+	err = yaml.Unmarshal(data, &ociRepository)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling OCIRepository file %s: %s", filename, err.Error())
```

Comment:
You can pass in `err` here without calling `.Error()` on it https://pkg.go.dev/fmt

> If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %x %X), or is %v but not %#v, the following two rules apply:
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).